### PR TITLE
Add missing events for working-group pallet.

### PR DESCRIPTION
**Pull request by shamil-gadelshin**
#### Changes
- add `RewardPaid` event
- add `WorkerStartedLeaving` event
- fix `leaving period` bug

### CODE REVIEW

**Review by bedeho — CHANGES_REQUESTED** [review]
LGTM. But don't we also need events for failed payments? This is currently a silent side-effect.

**Comment by Lezek123** [comment]
> LGTM. But don't we also need events for failed payments? This is currently a silent side-effect.

Right, it would be nice to have this information as part of the event/events too, as well as some information about whether the reward was paid as part of standard worker payout or was paid to "make up" for the previously missed reward.

**Comment by shamil-gadelshin** [comment]
Added new events:
- MissedRewardPaid 
- NewMissedRewardLevelReached

File: runtime-modules/working-group/src/lib.rs [hunk 1]
```diff
@@ -1376,7 +1432,7 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
         WorkerById::<T, I>::iter()
             .filter_map(|(worker_id, worker)| {
                 if let Some(started_leaving_at) = worker.started_leaving_at {
-                    if started_leaving_at + worker.job_unstaking_period >= Self::current_block() {
+                    if started_leaving_at + worker.job_unstaking_period <= Self::current_block() {
```
Comment (conectado) [inline runtime-modules/working-group/src/lib.rs:1435]: I don't understand this change, isn't the unstaking period finished after `job_unstaking_period`?

**Review by conectado — COMMENTED** [review]
I just left one question that I didn't understand a change.

File: runtime-modules/working-group/src/lib.rs (hunk 1 above)
Comment (shamil-gadelshin) [reply runtime-modules/working-group/src/lib.rs:1435]: It's easier to reason about if we rewrite the condition: `if current_block >= last_worker_date`

Comment (conectado) [reply runtime-modules/working-group/src/lib.rs:1435]: But this condition was correct before? It seems that before it was `if current_block <= worker_leaving_block` or there was some other change related to this?

Comment (shamil-gadelshin) [reply runtime-modules/working-group/src/lib.rs:1435]: Added an additional test to verify correct work of the `job_unstaking_period` parameter.

**Review by conectado — APPROVED** [review]
LGTM

**Pull request merged**